managed/models/database.go

```diff
@@ -1149,79 +1168,103 @@ func SetupDB(ctx context.Context, sqlDB *sql.DB, params SetupDBParams) (*reform.
+	return dbEncryption(tx, database, itemsToEncrypt, encryption.EncryptItems, encryptionExists, addToEncryptedItems)
+}
+
+func encryptionExists(m map[string]bool, key string) bool {
+	return m[key]
+}
+
+func addToEncryptedItems(encryptedItems []string, items []string) []string {
+	return slices.Concat(encryptedItems, items)
+}
+
+// DecryptDB decrypts a set of columns in a specific database and table.
+func DecryptDB(tx *reform.TX, database string, itemsToEncrypt []encryption.Table) error {
+	return dbEncryption(tx, database, itemsToEncrypt, encryption.DecryptItems, encryptionNotExists, removeFromEncryptedItems)
+}
+
+func encryptionNotExists(m map[string]bool, key string) bool {
+	return !encryptionExists(m, key)
+}
+
+func removeFromEncryptedItems(encryptedItems []string, items []string) []string {
+	res := []string{}
+	for _, encryptedItem := range encryptedItems {
+		exists := false
+		for _, item := range items {
+			if encryptedItem == item {
+				exists = true
+			}
+		}
+
+		if exists {
+			continue
+		}
+
+		res = append(res, encryptedItem)
+	}
+
+	return res
+}
+
+func dbEncryption(tx *reform.TX, database string, items []encryption.Table,
+	encryptionHandler func(tx *reform.TX, tables []encryption.Table) error,
+	checkHandler func(m map[string]bool, key string) bool,
+	settingsHandler func(encryptedItems []string, items []string) []string,
+) error {
+	if len(items) == 0 {
 		return nil
 	}
 
 	settings, err := GetSettings(tx)
 	if err != nil {
 		return err
 	}
-	alreadyEncrypted := make(map[string]bool)
+	currentColumns := make(map[string]bool)
 	for _, v := range settings.EncryptedItems {
-		alreadyEncrypted[v] = true
+		currentColumns[v] = true
 	}
 
-	notEncrypted := []encryption.Table{}
-	newlyEncrypted := []string{}
-	for _, table := range itemsToEncrypt {
+	tables := []encryption.Table{}
+	prepared := []string{}
+	for _, table := range items {
 		columns := []encryption.Column{}
 		for _, column := range table.Columns {
-			dbTableColumn := fmt.Sprintf("%s.%s.%s", params.Name, table.Name, column.Name)
-			if alreadyEncrypted[dbTableColumn] {
+			dbTableColumn := fmt.Sprintf("%s.%s.%s", database, table.Name, column.Name)
+			if checkHandler(currentColumns, dbTableColumn) {
 				continue
 			}
 
 			columns = append(columns, column)
-			newlyEncrypted = append(newlyEncrypted, dbTableColumn)
+			prepared = append(prepared, dbTableColumn)
 		}
 		if len(columns) == 0 {
 			continue
 		}
 
 		table.Columns = columns
-		notEncrypted = append(notEncrypted, table)
+		tables = append(tables, table)
 	}
-
-	if len(notEncrypted) == 0 {
+	if len(tables) == 0 {
 		return nil
 	}
 
-	err = encryption.EncryptItems(tx, notEncrypted)
+	err = encryptionHandler(tx, tables)
 	if err != nil {
 		return err
 	}
 	_, err = UpdateSettings(tx, &ChangeSettingsParams{
-		EncryptedItems: slices.Concat(settings.EncryptedItems, newlyEncrypted),
+		EncryptedItems: settingsHandler(settings.EncryptedItems, prepared),
```

Comment:
I think it can be simplified based on `expectedState` it can be `prepared` or empty slice.